cmd/horologium/main.go

```diff
@@ -199,14 +202,27 @@ func sync(prowJobClient ctrlruntimeclient.Client, cfg *config.Config, cr cronCli
 			}
 			continue
 		}
-		if !shouldTrigger {
+		if !shouldTrigger && p.RetriggerFailedRun == nil {
 			logger.WithFields(logrus.Fields{
 				"previous-found": previousFound,
 				"name":           p.Name,
 				"job":            p.JobBase.Name,
 			}).Debug("Trigger time has not yet been reached.")
 		}
+		run := false
+
+		// Proceed only if the run deviates from the normal flow.
+		if !shouldTrigger {
+			run = shouldTriggerFailedRun(j, p, now, logger, previousFound)
+		}
 		if !previousFound || shouldTrigger {
+			run = true
```

Comment:
can't this be simplified into this:
```suggestion
		if !previousFound || shouldTrigger || shouldTriggerFailedRun(j, p, now, logger, previousFound){
```